pkg/imgutil/filtering.go

```diff
@@ -145,18 +145,34 @@ func FilterByCreatedAt(ctx context.Context, client *containerd.Client, before []
 			maxTime = time.Now()
 		)
 
+		parseImageName := func(names []string) string {
+			var name string
+			if len(names) > 1 {
+				name = names[1]
+			} else {
+				name = names[0]
+			}
+			ind := strings.Index(name, "==")
+			if ind > 0 {
+				name = name[ind+2:]
+			}
+			return name
+		}
+
 		imageStore := client.ImageService()
 		if len(before) > 0 {
 			beforeImages, err := imageStore.List(ctx, before...)
 			if err != nil {
 				return []images.Image{}, err
 			}
-			if len(beforeImages) > 0 {
-				maxTime = beforeImages[0].CreatedAt
-				for _, image := range beforeImages {
-					if image.CreatedAt.After(maxTime) {
-						maxTime = image.CreatedAt
-					}
+			if len(beforeImages) == 0 {
+				//nolint:stylecheck
+				return []images.Image{}, fmt.Errorf("No such image: %s", parseImageName(before))
```

Comment:
why do we need `nolint` here?

Reply:
"No such image: aaaa:latest" is the output of docker. If don't add this `nolint`, then golint-ci won't pass, or we can change the output.

Reply:
>  If don't add this nolint, then golint-ci won't pass

what's the error message from golint-ci? (Or you mean `"No such image: aaaa:latest" is the output of docker.` is the error message from golint-ci?)

Reply:
```
 Running [/home/runner/golangci-lint-1.61.0-linux-amd64/golangci-lint run  --verbose] in [/home/runner/work/nerdctl/nerdctl] ...
  Error: pkg/imgutil/filtering.go:163:30: ST1005: error strings should not be capitalized (stylecheck)
  				return []images.Image{}, fmt.Errorf("No such image: %s", fetchImageNames(before))
  				                         ^
  Error: pkg/imgutil/filtering.go:179:30: ST1005: error strings should not be capitalized (stylecheck)
  				return []images.Image{}, fmt.Errorf("No such image: %s", fetchImageNames(since))
  	
```